### PR TITLE
Upgrade home view components to composition api

### PR DESCRIPTION
**Pull request by MatthewCai2002**
*Issue #:*
https://github.com/bcgov/entity/issues/<Put the github issue number here>

*Description of changes:*
    - updated, BCSCPanel, InfoStepper, LoginBCSC, NRbtn, TestimonialQuotes components to composition api
    - created unit tests for each component

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Comment by MatthewCai2002** [comment]
important vue files are BCSCPanel.vue, InfoStepper.vue, LoginBCSC.vue, NameRequestButton.vue, TestimonialQuotes.vue, HomeView.vue
test files: BCSCPanel.spec.ts, InfoStepper.spec.ts, LoginBCSC.spec.ts, NameRequestButton.spec.ts, TestimonialQuotes.spec.ts, HomeView.spec.ts

**Comment by MatthewCai2002** [comment]
InfoStepper tests are really weird, wasn't able to get test utils to find elements with classes bound to them based on whether or not a step was selected. this prevented testing the stepping behaviour. I commented out those tests

File: auth-web/src/components/auth/home/BcscPanel.vue [hunk 1]
```diff
@@ -52,36 +52,50 @@
 </template>
 
 <script lang="ts">
-import { Component, Emit, Prop, Vue } from 'vue-property-decorator'
+import { PropType, defineComponent } from '@vue/composition-api'
 import LearnMoreButton from '@/components/auth/common/LearnMoreButton.vue'
 import { User } from '@/models/user'
+import Vue from 'vue'
```
Comment (kialj876) [inline auth-web/src/components/auth/home/BcscPanel.vue:58]: unused import

File: auth-web/src/components/auth/home/BcscPanel.vue [hunk 2]
```diff
@@ -40,8 +40,8 @@
     </v-row>
     <!-- Panel Btns -->
     <div class="mt-10">
-      <template v-if="!userProfile">
-        <v-btn large color="bcgovblue" class="cta-btn font-weight-bold white--text mr-2 px-7"
+      <template v-if="!user">
+        <v-btn large color="bcgovblue" class="cta-btn font-weight-bold white--text mr-2 px-7 "
```
Comment (kialj876) [inline auth-web/src/components/auth/home/BcscPanel.vue:44]: remove extra space pls

File: auth-web/src/components/auth/home/BcscPanel.vue [hunk 3]
```diff
@@ -52,36 +52,50 @@
...
+    user: Object as PropType<User>
+  },
+  emits: ['login', 'account-dialog'],
+  setup (props, { emit }) {
+    const cardSetUpUrl = 'https://www2.gov.bc.ca/gov/content/governments/government-id/bcservicescardapp/setup'
+    const learnMoreUrl = 'https://www2.gov.bc.ca/gov/content/governments/government-id/bcservicescardapp/setup'
+    const secureBulletPoints = [
+      { text: 'A mobile card is a representation of your BC Services Card on your mobile device. It\'s used to prove who you are when you log in to access government services online.' },
+      { text: 'Only your name and a unique identifier is stored on the mobile device.' }
+    ] as { text: String }[]
```
Comment (kialj876) [inline auth-web/src/components/auth/home/BcscPanel.vue:75]: remove 'as...' The typing will be inferred automatically for this

File: auth-web/src/components/auth/home/BcscPanel.vue [hunk 4]
```diff
@@ -52,36 +52,50 @@
...
+
+    const easeBulletPoints = [
+      { text: 'You can verify your identity by video right from your mobile device. You don\'t need to go in person unless you can\'t verify by video.' }
+    ] as { text: String }[]
```
Comment (kialj876) [inline auth-web/src/components/auth/home/BcscPanel.vue:79]: same thing here

File: auth-web/src/components/auth/home/InfoStepper.vue [hunk 5]
```diff
@@ -32,60 +32,75 @@
...
-      text: 'Register or Incorporate',
-      to: '/home/incorporate-or-register'
-    },
-    {
-      id: 'step-4-btn',
-      step: 4,
-      text: 'Maintain Your Business',
-      to: '/home/maintain-business'
+import { computed, defineComponent } from '@vue/composition-api'
+import Vue from 'vue'
```
Comment (kialj876) [inline auth-web/src/components/auth/home/InfoStepper.vue:36]: unused import

File: auth-web/src/components/auth/home/InfoStepper.vue [hunk 6]
```diff
@@ -32,60 +32,75 @@
...
+      }
+    ]
+
+    const getCurrentStep = (): number => {
+      const route = root.$route?.path
+      for (const path of steps) {
+        if (path.to === route) {
+          return path.step || 0
+        }
+      }
```
Comment (kialj876) [inline auth-web/src/components/auth/home/InfoStepper.vue:74]: should probably have a `return 0` at the bottom here in case`root.$route?.path` is undefined?

File: auth-web/src/components/auth/home/InfoStepper.vue [hunk 7]
```diff
@@ -32,60 +32,75 @@
...
-    this.$router.push(nextStep.to)
-  }
+    const isCurrentStep = (step: any): boolean => {
+      return root.$route?.path === step.to
+    }
 
-  private isCurrentStep (step: any): boolean {
-    return this.$route.path === step.to
-  }
+    const goTo = (step: any): void => {
```
Comment (kialj876) [inline auth-web/src/components/auth/home/InfoStepper.vue:85]: can you please type the `step` arg here?

File: auth-web/src/components/auth/home/LoginBCSC.vue [hunk 8]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <v-card>
+  <v-card class="test-class">
```
Comment (kialj876) [inline auth-web/src/components/auth/home/LoginBCSC.vue:2]: I don't think you meant to leave this in?

File: auth-web/src/components/auth/home/LoginBCSC.vue [hunk 9]
```diff
@@ -19,14 +19,26 @@
...
+    return {
+      bcscRedirectUrl,
+      login
+    }
   }
-}
+})
 </script>
+<style lang="scss" scoped>
+  .test-class {
```
Comment (kialj876) [inline auth-web/src/components/auth/home/LoginBCSC.vue:41]: please remove if this is not supposed to be in?

File: auth-web/src/components/auth/home/LoginBCSC.vue [hunk 10]
```diff
@@ -19,14 +19,26 @@
...
 <script lang="ts">
-import { Component, Vue } from 'vue-property-decorator'
-@Component({
-  name: 'LoginBCSC'
-})
-export default class LoginBCSC extends Vue {
-  private readonly bcscRedirectUrl = 'https://www2.gov.bc.ca/gov/content/governments/government-id/bc-services-card/log-in-with-card'
-  private login () {
-    this.$router.push('/signin/bcsc/createaccount')
+import Vue from 'vue'
```
Comment (kialj876) [inline auth-web/src/components/auth/home/LoginBCSC.vue:22]: unused import

File: auth-web/src/components/auth/home/NameRequestButton.vue [hunk 11]
```diff
@@ -12,27 +12,40 @@
 </template>
 
 <script lang="ts">
-import { Component, Prop, Vue } from 'vue-property-decorator'
 import ConfigHelper from '@/util/config-helper'
 import { LDFlags } from '@/util/constants'
 import LaunchDarklyService from 'sbc-common-components/src/services/launchdarkly.services'
 import { appendAccountId } from 'sbc-common-components/src/util/common-util'
+import { defineComponent } from '@vue/composition-api'
+import vue from 'vue'
```
Comment (kialj876) [inline auth-web/src/components/auth/home/NameRequestButton.vue:20]: unused import

File: auth-web/src/components/auth/home/NameRequestButton.vue [hunk 12]
```diff
@@ -12,27 +12,40 @@
...
-@Component({})
-export default class NameRequestButton extends Vue {
-  @Prop() isWide: boolean
-
-  @Prop({ default: false }) isInverse: boolean
+export default defineComponent({
+  name: 'NameRequestButton',
+  props: {
+    isWide: {
+      type: Boolean,
```
Comment (kialj876) [inline auth-web/src/components/auth/home/NameRequestButton.vue:26]: you can remove the type here as it will infer it from your default

File: auth-web/src/components/auth/home/NameRequestButton.vue [hunk 13]
```diff
@@ -12,27 +12,40 @@
...
+      default: false
+    },
+    isInverse: {
+      type: Boolean,
```
Comment (kialj876) [inline auth-web/src/components/auth/home/NameRequestButton.vue:30]: same here

File: auth-web/src/components/auth/home/TestimonialQuotes.vue [hunk 14]
```diff
@@ -9,27 +9,53 @@
   </section>
 </template>
 <script lang="ts">
-import { Component, Vue } from 'vue-property-decorator'
+import { defineComponent, onMounted, ref } from '@vue/composition-api'
+import vue from 'vue'
```
Comment (kialj876) [inline auth-web/src/components/auth/home/TestimonialQuotes.vue:13]: unused import

File: auth-web/src/components/auth/home/TestimonialQuotes.vue [hunk 15]
```diff
@@ -9,27 +9,53 @@
...
+  name: 'TestimonialQuotes',
+  setup () {
+    const quoteIndex = ref(0)
+    const quotes = [
       {
         text: 'We are really excited by the idea of incorporating as a Benefit Company, it really makes sense for us and the way we want to run our business.',
         author: 'Business Founder'
       }
-    ]
+    ] as { text: string, author: string}[]
```
Comment (kialj876) [inline auth-web/src/components/auth/home/TestimonialQuotes.vue:24]: dont need typing here as it will infer it

File: auth-web/src/components/auth/home/TestimonialQuotes.vue [hunk 16]
```diff
@@ -9,27 +9,53 @@
...
+    onMounted(setInterval)
+    return {
+      quoteIndex,
+      quotes
+    }
   }
-}
+})
+
+// @Component({})
```
Comment (kialj876) [inline auth-web/src/components/auth/home/TestimonialQuotes.vue:40]: remove commented out code

File: auth-web/src/components/auth/home/TestimonialQuotes.vue [hunk 17]
```diff
@@ -9,27 +9,53 @@
...
       {
         text: 'We are really excited by the idea of incorporating as a Benefit Company, it really makes sense for us and the way we want to run our business.',
         author: 'Business Founder'
       }
-    ]
+    ] as { text: string, author: string}[]
 
-  mounted () {
-    setInterval(() => {
+    const setInterval = () => {
```
Comment (kialj876) [inline auth-web/src/components/auth/home/TestimonialQuotes.vue:26]: this was a built in Window function I think? It's supposed to execute it every 5 seconds

File: auth-web/src/views/auth/home/HomeView.vue [hunk 18]
```diff
@@ -51,13 +51,13 @@
             </div>
           </div>
 
+          <LoginBCSC>
```
Comment (kialj876) [inline auth-web/src/views/auth/home/HomeView.vue:54]: why did you move this out of the dialog?

File: auth-web/tests/unit/components/BcscPanel.spec.ts [hunk 19]
```diff
@@ -56,57 +54,31 @@ describe('BcscPanel.vue', () => {
...
-    expect(createAccountLink).toBeDefined()
-    expect(createAccountLink.textContent).toContain('Create a BC Registries Account')
+    expect(createAccountLink.exists()).toBe(true)
+    expect(createAccountLink.text()).toContain('Create a BC Registries Account')
 
     expect(authenticatedBtns[1]).toBeDefined()
     expect(authenticatedBtns[1].textContent).toContain('Learn More')
   })
-
-  it('renders the correct text and number of bullet points', async () => {
```
Comment (kialj876) [inline auth-web/tests/unit/components/BcscPanel.spec.ts:92]: why are you removing this test? Is it no longer valid?

File: auth-web/tests/unit/components/InfoStepper.spec.ts [hunk 20]
```diff
@@ -0,0 +1,98 @@
...
+    expect(wrapper.find('#step-buttons-container').exists()).toBe(true)
+    expect(wrapper.find('.next-step-btn').exists()).toBe(true)
+    expect(wrapper.find('.step').exists()).toBe(true)
+  })
+
+  it('renders correct number of steps', () => {
+    expect(wrapper.findAll('.step').length).toBe(4)
+  })
+
+//     it('selects first step on render', async () => {
```
Comment (kialj876) [inline auth-web/tests/unit/components/InfoStepper.spec.ts:72]: how come all these are commented out? They look like good scenarios to test

**Review by kialj876 — COMMENTED** [review]
Lots of really good work in here Mathew! I have a lot of comments, but they are all minor things

File: auth-web/tests/unit/components/InfoStepper.spec.ts (hunk 20 above)
Comment (kialj876) [reply auth-web/tests/unit/components/InfoStepper.spec.ts:72]: its probably because root.$route.path is not evaluating properly in your test. You could try mocking that and the push out in your test or verify that thats being set / the things using that are being set correctly

File: auth-web/src/components/auth/home/BcscPanel.vue [hunk 21]
```diff
@@ -52,36 +52,49 @@
...
+    const secureBulletPoints = [
+      { text: 'A mobile card is a representation of your BC Services Card on your mobile device. It\'s used to prove who you are when you log in to access government services online.' },
+      { text: 'Only your name and a unique identifier is stored on the mobile device.' }
+    ]
+
+    const easeBulletPoints = [
+      { text: 'You can verify your identity by video right from your mobile device. You don\'t need to go in person unless you can\'t verify by video.' }
+    ]
+
+    const emitLogin = () => {
```
Comment (kialj876) [inline auth-web/src/components/auth/home/BcscPanel.vue:80]: is this actually used at all? I think you can remove it?

File: auth-web/src/components/auth/home/BcscPanel.vue [hunk 22]
```diff
@@ -52,36 +52,49 @@
...
+      emit('login')
+    }
+
+    const emitAccountDialog = () => {
```
Comment (kialj876) [inline auth-web/src/components/auth/home/BcscPanel.vue:84]: same with this I think?

File: auth-web/tests/unit/components/BcscPanel.spec.ts (hunk 19 above)
Comment (kialj876) [reply auth-web/tests/unit/components/BcscPanel.spec.ts:92]: ah I see its because the value isn't reactive anymore so the test setup is invalid. Can you just alter it so it still verifies the static bullet points?

**Comment by seeker25** [comment]
Package / package lock.json. I'd also like to take a look at this before it's merged please.

**Comment by MatthewCai2002** [comment]
package/package.json weren't really changed but I updated the version from 1.1.52 to 1.1.53

File: auth-web/tests/unit/components/BcscPanel.spec.ts (hunk 19 above)
Comment (kialj876) [reply auth-web/tests/unit/components/BcscPanel.spec.ts:92]: -- resolving since you have one testing that above
